File: linked-lists/simplify-path/simplify-path.ts
function simplifyPath(path: string): string {
  const list: string[] = [];

  const dirs = path.split('/');

  for (let i = 0; i < dirs.length; i++) {
    if (dirs[i] === '..') {
      if (list.length > 0) {
        list.pop();
      }
    } else if (dirs[i] !== '' && dirs[i] !== '.') {
      list.push(dirs[i]);
    }
  }

  if (list.length === 0) {
    return '/';
  }

  return '/' + list.join('/');
}

File: linked-lists/simplify-path/simplify-path.go
package simplifypath

import "strings"

func simplifyPath(path string) string {
	list := []string{}

	for _, dir := range strings.Split(path, "/") {
		if dir == ".." {
			if len(list) > 0 {
				list = list[:len(list)-1]
			}
		} else if dir != "" && dir != "." {
			list = append(list, dir)
		}
	}

	if len(list) == 0 {
		return "/"
	}

	return "/" + strings.Join(list, "/")
}
